core/lib/generators/solidus/install/install_generator.rb

```diff
@@ -322,28 +323,33 @@ def detect_authentication_to_install
     def detect_payment_method_to_install
       return 'paypal' if Bundler.locked_gems.dependencies['solidus_paypal_commerce_platform']
       return 'bolt' if Bundler.locked_gems.dependencies['solidus_bolt']
+      return 'braintree' if Bundler.locked_gems.dependencies['solidus_braintree']
 
-      descriptions = {
-        paypal: "- [#{set_color 'paypal', :bold}] Install `solidus_paypal_commerce_platform` (#{set_color :default, :bold}).",
-        bolt: "- [#{set_color 'bolt', :bold}] Install `solidus_bolt`.",
-        none: "- [#{set_color 'none', :bold}] Skip installing a payment method.",
+      allowable_payment_methods_hash = {
+        'none' => ['paypal', 'braintree', 'none'],
+        'classic' => ['paypal', 'bolt', 'none'],
+        'starter' => ['paypal', 'braintree', 'none'],
       }
 
-      payment_methods = PAYMENT_METHODS
+      payment_methods_hash = {
+        'paypal' => "- [#{set_color 'paypal', :bold}] Install `solidus_paypal_commerce_platform` (#{set_color :default, :bold}).",
+        'bolt' => "- [#{set_color 'bolt', :bold}] Install `solidus_bolt`.",
+        'braintree' => "- [#{set_color 'braintree', :bold}] Install `solidus_braintree`.",
+        'none' => "- [#{set_color 'none', :bold}] Skip installing a payment method.",
+      }
 
-      if @selected_frontend != 'classic'
-        payment_methods -= ['bolt']
-        descriptions.delete(:bolt)
+      payment_methods_hash.select! do |payment_method, _|
```

Comment:
What do you think about using the non-mutable version (`select`, without the `!`) instead? I know would require naming another variable (`selected_payment_methods` ?), but mutation can always bring surprises later. That's a non-blocking, as I understand that could be something opinionated.

Reply:
Yeah, I agree that generally using non-mutable methods is better. I'll update the code.